turbopy/core.py

```diff
@@ -463,6 +469,9 @@ def __init__(self, owner: Simulation, input_data: dict):
         self.owner = owner
         self.input_data = input_data
         self.name = input_data["type"]
+        self.custom_name = None
+        if "instance" in input_data:
```

Comment:
Why are you looking for `"instance"` instead of something more specific... like `"custom name"`?